common/unsafe/src/main/java/org/apache/spark/sql/catalyst/expressions/HiveHasher.java

```diff
@@ -38,6 +39,10 @@ public static int hashLong(long input) {
     return (int) ((input >>> 32) ^ input);
   }
 
+  public static int hashUnsafeBytesBlock(MemoryBlock base, long offset, int lengthInBytes) {
```

Comment:
why we call it `hashUnsafeBytesBlock` while passing in a general `MemoryBlock`?

Reply:
This is based on [this discussion](https://github.com/apache/spark/pull/19222#discussion_r138744794). Currently, when I can see large performance improvement, I do not call non-MemoyBlock version.